### ISSUE TITLE
One-Hot encoding

### ISSUE THREAD

**Issue by wellknownram** [YazanIST/Diabetes-Classification-Model#1]
Hi, speaking without much experience but did you consider using one-hot encoding instead of replacing the categories with numerical values ? because from what I know, it might give the model a false sense of magnitude or ranking.

`from sklearn.preprocessing import OneHotEncoder

labels = ['never', 'current', 'former', 'ever', 'not current']
encoder = OneHotEncoder(sparse=False, categories=[labels])
encoded_data = encoder.fit_transform(data[['smoking_history']])

data[['smoking_history']] = encoded_data`

this code might do the job with some adjustments :)

**Comment by YazanIST** [comment]
I did the notebook previously without much experience just for trying
things for a course, I am redoing the models and notebook now for a paper
on a higher level and more accurate details, if you’re interested I can
keep u updated:)

**Comment by wellknownram** [comment]
Would like that !
